deploy/helm/kubecf/templates/_config.tpl

```diff
@@ -0,0 +1,263 @@
+|
+| When the found value is `nil`, we still return the empty string and not a stringified
+| nil (which would be "<no data>"), because helm prints it as an empty string, but
+| otherwise treats as a non-empty string for all other purposes.
+|
+| $kubecf.retval is also set to the found entry in case the caller needs an object and
+| not just a string. This can also be used to disambiguate between nil and the empty
+| string (which can mean: not found).
+|
+| When $context is an array, then _lookup will look for an array element that has a "name"
+| property matching the first element of $path.
+|
+| Example: Looking for "instance_groups.api.jobs.cloud_controller_ng" in $.kubecf.manifest
+| is equivalent to the "/instance_groups/name=api/jobs/name=cloud_controller_ng" path in
+| JSON-patch.
++-----------------------------------------------------------------------------------------
+| Not implemented:
+|
+| When the full $path cannot be found, and the $.kubecf.required variable is true, then
+| _lookup should throw an error instead of returning an empty string. It could use
+| $.kubecf.query to provide a helpful error message. With the introduction of a
+| validation schema, this may no longer be required.
+|
+| There is no mechanism to search array $context for any other properties but "name".
+| It can easily be done on the caller side as well, but otherwise should be easy to
+| add here as well. So far we don't need it.
+==========================================================================================
+*/}}
+{{- define "_config._lookup" }}
+  {{- $kubecf := index . 0 }}
+  {{- $context := index . 1 }}
+  {{- $path := slice . 2 }}
+  {{- $key := (first $path) }}
+  {{- if kindIs "slice" $context }}
+    {{- $_ := set $kubecf "retval" nil }}
+    {{- range $context }}
+      {{- if eq (index . "name") $key }}
+        {{- $_ := set $kubecf "retval" . }}
+      {{- end }}
+    {{- end }}
+  {{- else }}
+    {{- $_ := set $kubecf "retval" (index $context $key) }}
+  {{- end }}
+  {{- if $kubecf.retval }}
+    {{- if gt (len $path) 1 }}
+      {{- include "_config._lookup" (concat (list $kubecf $kubecf.retval) (rest $path)) }}
+    {{- else }}
+      {{- $kubecf.retval }}
+    {{- end }}
+  {{- end }}
+{{- end }}
+
+
+{{- /*
+==========================================================================================
+| _config.condition (list $ $condition)
++-----------------------------------------------------------------------------------------
+| Evaluates $condition and returns either the string "true" or the empty string.
+|
+| - A nil (missing) condition is always true.
+|
+| - A boolean condition is the value of the $condition itself.
+|
+| - Otherwise $condition must be a string. All spaces will be removed first.
+|
+| - The string can be a list of "," separated OR terms. The condition is true
+|   if any of the terms are true.
+|
+| - An OR term can be a list of "&" separated AND terms. The OR term is true
```

Comment:
`&` is usually used for bit masking. I find it confusing when used in a DSL because people just don't know the DSL and they try to infer the meaning of it without looking into documentation (when it exists). For this reason, I'm in favour of (`AND` and `OR`) or literal (`&&` and `||`). If you don't remove spaces before doing the lexing and parsing, it becomes easier to not do the splitting in the wrong place. I also know it's not easy to write a DSL in Go templating, so kudos to you! :smile:

Reply:
I may switch to `&&` and `||` while implementing support for parenthesis; we'll see. Waiting for a full code review before making any changes.

Reply:
Switched to `&&` and `||` and added parenthesis support in e895e02.